TourMate/TourMate/Managers/FirebaseAuthenticationManager.swift

```diff
@@ -70,12 +70,20 @@ struct FirebaseAuthenticationManager: AuthenticationManager {
         let credential = GoogleAuthProvider.credential(withIDToken: idToken, accessToken: authentication.accessToken)
 
         Auth.auth().signIn(with: credential) { authResult, error in
-            if let result = authResult,
-               let userAdditionalInfo = result.additionalUserInfo {
+
+            if let error = error {
+                print("[FirebaseAuthenticationManager] Firebase authentication failed: \(error.localizedDescription)")
+            } else if let result = authResult {
```

Comment:
The logic for if-else feels a little confusing, can we change it to return after the error, and then using a guard clause to check the result?

Reply:
Yes sounds good, will make the change